### PR TITLE
Pr/player action packet

### PR DESCRIPTION
**Pull request by FrederoxDev**

### CODE REVIEW

**Comment by Adrian8115** [comment]
Istg I was about the push it myself

File: proto/src/packets/player_action.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+use bedrockrs_core::int::{VAR, LE};
+use bedrockrs_proto_derive::ProtoCodec;
+use crate::types::network_block_pos::NetworkBlockPos;
+use bedrockrs_shared::actor_runtime_id::ActorRuntimeID;
+
+#[derive(ProtoCodec, Debug, Clone)]
+pub struct PlayerActionPacket {
+    pub player_runtime_id: ActorRuntimeID,
+    pub action: VAR<i32>,
```
Comment (theaddonn) [inline proto/src/packets/player_action.rs:51]: could you turn this into an enum?

File: proto/src/packets/level_chunk.rs [hunk 2]
```diff
@@ -38,6 +38,7 @@ impl ProtoCodec for LevelChunkPacket {
             unimplemented!()
         }
 
+        VAR::<u32>::new(self.serialized_chunk_data.len().try_into().unwrap()).proto_serialize(stream)?;
```
Comment (theaddonn) [inline proto/src/packets/level_chunk.rs:41]: unwrap makes me upset :c

this code here does the same without unwrap
```rust
let len = self.serialized_chunk_data
   .len()
   .try_into()
   .map_err(|e| ProtoCodecError::FromIntError(e))?;

VAR::<u32>::new(len).proto_serialize(stream)?;
```

**Review by Adrian8115 — CHANGES_REQUESTED** [review]

File: proto/src/packets/player_action.rs (hunk 1 above)
Comment (FrederoxDev) [reply proto/src/packets/player_action.rs:51]: Idk what the fields r, if ya were about to push this, maybe just close pr for now and u push ur stuff

Comment (theaddonn) [reply proto/src/packets/player_action.rs:51]: I am working on a similar packet, just confused them...
tho here are the possible types:
image
you can create a simple enum with them and use the `ProtoCodec` derive macro

**Comment by Adrian8115** [comment]
I will upstream those changes later manually, thanks for all the help!

**Pull request closed without merging**